### ISSUE TITLE
High Priority - WCAG Level A - Non-descriptive links

### ISSUE THREAD

**Issue by JuliaSang** [w3c/w3c-website#161]
The language selector links were problematic for screen reader users.

WCAG Reference(s):
2.4.4 Link Purpose – in context (Level A)
[Understanding Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html) | [How to Meet Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/quickref/#link-purpose-in-context)

2.4.9 Link Purpose - Link Only - (Level AAA)
[Understanding Link Purpose (Link Only)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-link-only.html) | [How to Meet Link Purpose (Link Only)](https://www.w3.org/WAI/WCAG21/quickref/#link-purpose-link-only)

**Issue ID: DAC_NonDescriptive_Links_01**

URL: https://beta.w3.org/
Page title: We believe in one web for all – W3C 

Journey 1
    
<img width="408" alt="mobile language selector" src="https://user-images.githubusercontent.com/20167528/159244402-da4e269b-84b7-45d9-a5b8-6b80feffb397.png">

Jaws link list

<img width="572" alt="JAWS links list" src="https://user-images.githubusercontent.com/20167528/159244409-9af383d5-5913-44ff-8efc-fe8872f0ba59.png">

<img width="807" alt="language-selector" src="https://user-images.githubusercontent.com/20167528/159244412-016ce72b-686e-4b1c-ab92-71001169554a.png">

The language selector links are ambiguous for screen reader users browsing out of context on desktop using JAWS, and do not announce either in or out of context for NVDA users.

Screen reader users that are browsing out of context using both JAWS and NVDA using the links list dialogue hear the links announced as '3 of 36' and '4 of 36'. Screen reader users will be unable to determine the purpose of the links when navigating in this way on desktop.

While browsing in context using NVDA both links are relayed as “Link”. This appears to be an issue with NVDA which is preventing the language from being detected correctly.

VoiceOver and TalkBack announce the links in Japanese and Chinese. 

**Current code ref(s):** #lang-nav > div > div > ul

```
<ul class="clean-list" role="list">
<li><a href="/ja/" hreflang="ja" lang="ja">日本語ホームページ</a></li>
<li><a href="/zh-hans/" hreflang="zh-hans" lang="zh-hans">简体中文首页</a></li> </ul>
```

**Solution:**
Whilst NVDA does indicate that it supports Japanese and Chinese (simplified and traditional), there may be a conflict in the language settings which is preventing it from relaying the information to users; however some additional context needs to be provided to ensure that the language selector links do not announce simply as 'Link' for users who navigate using NVDA both in and out of context, and JAWS out of context.

Whilst the links are read correctly in some other instances, to ensure that the information is clear to users regardless of the screen reading software being used consider including visually hidden text to the link text using an '[sr-only](url)' class and position this after the current link text. For example 'Go to Japanese site'.

**Please note:** where English text is added to a foreign language link, the language attribute will need to be specified accordingly so that both parts of the link are read correctly.

**Comment by maries24** [comment]
Solution to implement:

> to ensure that the information is clear to users regardless of the screen reading software being used consider including visually hidden text to the link text using an '[sr-only](https://github.com/w3c/w3c-website/issues/url)' class and position this after the current link text. For example 'Go to Japanese site'.

> Please note: where English text is added to a foreign language link, the language attribute will need to be specified accordingly so that both parts of the link are read correctly.

**Comment by NicolaSaunders** [comment]
@maries24 Please note that the classname in the design system to use for the hidden text is `visuallyhidden`
